cmd/create-wallet.go

```diff
@@ -0,0 +1,70 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/0chain/gosdk/zcncore"
+	"github.com/spf13/cobra"
+)
+
+var createWalletCmd = &cobra.Command{
+	Use:   "create-wallet",
+	Short: "Create wallet and logs it into stdout (pass --register to register wallet to blockchain)",
+	Long:  `Create wallet and logs it into standard output (pass --register to register wallet to blockchain)`,
+	Args:  cobra.MaximumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		online, err := cmd.Flags().GetBool("register")
+		if err != nil {
+			ExitWithError("invalid register flag")
+		}
+
+		var walletStr string
+		if online {
+			statusBar, err := createWallet()
+			if err != nil {
+				ExitWithError(fmt.Printf("Failed to create wallet: %v", err))
+			}
+			walletStr = statusBar.walletString
+
+			// update gosdk wallet to use the newly generate wallet for following operation
+			if err = zcncore.SetWalletInfo(walletStr, false); err != nil {
+				ExitWithError("failed to use new wallet", err)
+			}
+
+			log.Print("Creating related read pool for storage smart-contract...")
+			if err := createReadPool(); err != nil {
+				log.Fatalf("Failed to create read pool: %v", err)
+			}
+			log.Printf("Read pool created successfully")
+
+		} else {
+			walletStr, err = zcncore.CreateWalletOffline()
+			if err != nil {
+				ExitWithError("failed to generate offline wallet", err)
+			}
+		}
+
+		// write wallet into wallet dir
+		now := time.Now().UTC()
+		filename := filepath.Join(getConfigDir(),
+			fmt.Sprintf("%s_wallet_%x.json", now.Format("2006_01_02"), now.Unix()))
+
+		if err := os.WriteFile(filename, []byte(walletStr), 0644); err != nil {
```

Comment:
Please check if the wallet file with the same name exist before writing to it to avoid wallet to be overwritten.

Reply:
FYI - there won't be any name collision - as the hex suffix is the `hex(unix current timestamp)`. We can even opt for `now.UnixNano()` to further reduce the collision boundary (second vs nanosecond).

For, the user passed wallet name, yes we have to check and return an error.
Let's add it in both cases : )

Reply:
@bisakhmondal we need to check the name collision even the chance to get collision is very low as we can't accept the lose for wallet got overwritten.

Reply:
Done. 

```sh
./zwallet create-wallet  --silent
wallet saved in /home/bisakh/.zcn/2022_11_06_wallet_1724f8ff4135c2a1.json
```

```sh
./zwallet create-wallet  --wallet abc.json
0chain-core-sdk [INFO]   2022/11/06 10:45:59.732714 wallet_base.go:262: ******* Wallet SDK Version:v1.8.8-8-g3c6d3d6c ******* (SetLogFile)
0chain-core-sdk [INFO]   2022/11/06 10:46:00.016009 transaction_base.go:138: ******* Wallet SDK Version:v1.8.8-8-g3c6d3d6c ******* (InitZCNSDK)
wallet saved in /home/bisakh/.zcn/abc.json
{"client_id":"583c722a4eae3e06267a9fe6d5f850a84fec80c315e15c932f22c9e0f7810f43","client_key":"4e464cfc9441db3ca3f7f9b70b04df6e24422269001381e508119a73abea1804e5d1ce5440176470e30831a6f7a691a8271a1f1be3f40057ffe41b9f3ba72a86","keys":[{"public_key":"4e464cfc9441db3ca3f7f9b70b04df6e24422269001381e508119a73abea1804e5d1ce5440176470e30831a6f7a691a8271a1f1be3f40057ffe41b9f3ba72a86","private_key":"543aace593cc6f59c640e6d6c8d35c6a773b349a2e0b6ab4060f05c5b6bde613"}],"mnemonics":"code extend save cannon spawn biology enroll jewel april hello moral six extend please execute sing outdoor wire toast spray photo flash drive use","version":"1.0","date_created":"2022-11-06T10:46:00Z","nonce":0}%   
```

```sh
./zwallet create-wallet  --wallet abc.json --silent
unable to write wallet, file with "/home/bisakh/.zcn/abc.json" name already exists
```